Program.cs

```diff
@@ -88,6 +110,34 @@ internal static async Task<int> Main(string[] args)
 
                 var manifest = (Manifest?) serializer.Deserialize(reader) ?? throw new InvalidDataException();
 
+                bool matching = false;
+                for (int i = 0; i < compareManifests.Count; i++)

```

Comment:
This means we're parsing everything n^2 times, we can just pre-parse the thing we're comparing against. I'd also then put them into a HashSet after implementing GetHashCode (just use Hash.Combine). Then you can skip the entire looping bit for both perf and brevity.

Reply:
Hash combine `Name`, `Description`, `Dependencies`, `Link?`

Reply:
You'd want the elements of the dependencies instead of the array itself but yeah